### ISSUE TITLE
train my own dataset , but AP  is very low

### ISSUE THREAD

**Issue by ing-NT** [leoxiaobin/deep-high-resolution-net.pytorch#264]
Recently I  run HRnet keypoint detection code, I didn't use any PRETRAINED MODEL (including ImageNet and COCO datasets), then train and test（use groundtruth bbox） my own datasets(My own dataset is already in coco format) ，AP only around 1%. However, using  pretrained (coco trained in the paper), AP can reach about 88%. I don't know why, and do you know the  reason?@leoxiaobin
My configuration file is as follows：
AUTO_RESUME: true
CUDNN:
  BENCHMARK: true
  DETERMINISTIC: false
  ENABLED: true
DATA_DIR: ''
GPUS: (0,)
OUTPUT_DIR: 'output'
LOG_DIR: 'log'
WORKERS: 24
PRINT_FREQ: 100
DATASET:
  COLOR_RGB: true
  DATASET: 'coco'
  DATA_FORMAT: jpg
  FLIP: true
  NUM_JOINTS_HALF_BODY: 8
  PROB_HALF_BODY: 0.3
  ROOT: 'data/coco/'
  ROT_FACTOR: 45
  SCALE_FACTOR: 0.35
  TEST_SET: 'val2017'
  TRAIN_SET: 'train2017'
MODEL:
  INIT_WEIGHTS: true
  NAME: pose_hrnet
  NUM_JOINTS: 17
  PRETRAINED: ' '
  TARGET_TYPE: gaussian
  IMAGE_SIZE:
  - 192
  - 256
  HEATMAP_SIZE:
  - 48
  - 64
  SIGMA: 2
  EXTRA:
    PRETRAINED_LAYERS:
    - 'conv1'
    - 'bn1'
    - 'conv2'
    - 'bn2'
    - 'layer1'
    - 'transition1'
    - 'stage2'
    - 'transition2'
    - 'stage3'
    - 'transition3'
    - 'stage4'
    FINAL_CONV_KERNEL: 1
    STAGE2:
      NUM_MODULES: 1     
      NUM_BRANCHES: 2
      BLOCK: BASIC
      NUM_BLOCKS:
      - 4
      - 4
      NUM_CHANNELS:
      - 32
      - 64
      FUSE_METHOD: SUM
    STAGE3:
      NUM_MODULES: 4
      NUM_BRANCHES: 3
      BLOCK: BASIC
      NUM_BLOCKS:
      - 4
      - 4
      - 4
      NUM_CHANNELS:
      - 32
      - 64
      - 128
      FUSE_METHOD: SUM
    STAGE4:
      NUM_MODULES: 3
      NUM_BRANCHES: 4
      BLOCK: BASIC
      NUM_BLOCKS:
      - 4
      - 4
      - 4
      - 4
      NUM_CHANNELS:
      - 32
      - 64
      - 128
      - 256
      FUSE_METHOD: SUM
LOSS:
  USE_TARGET_WEIGHT: true
TRAIN:
  BATCH_SIZE_PER_GPU: 32
  SHUFFLE: true
  BEGIN_EPOCH: 0
  END_EPOCH: 210
  OPTIMIZER: adam
  LR: 0.001
  LR_FACTOR: 0.1
  LR_STEP:
  - 170
  - 200
  WD: 0.0001
  GAMMA1: 0.99
  GAMMA2: 0.0
  MOMENTUM: 0.9
  NESTEROV: false
TEST:
  BATCH_SIZE_PER_GPU: 32
  COCO_BBOX_FILE: 'data/coco/person_detection_results/COCO_val2017_detections_AP_H_56_person.json'
  BBOX_THRE: 1.0
  IMAGE_THRE: 0.0
  IN_VIS_THRE: 0.2
  MODEL_FILE: ''
  NMS_THRE: 1.0
  OKS_THRE: 0.9
  USE_GT_BBOX: true
  FLIP_TEST: true
  POST_PROCESS: true
  SHIFT_HEATMAP: true
DEBUG:
  DEBUG: true
  SAVE_BATCH_IMAGES_GT: true
  SAVE_BATCH_IMAGES_PRED: true
  SAVE_HEATMAPS_GT: true
  SAVE_HEATMAPS_PRED: true

**Comment by khandriod** [comment]
hi, 
how you used this architecture on a custom dataset?

Thanks

**Comment by ing-NT** [comment]
Make the same data set format as COCO2017 and put it in the same place as the code





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "leoxiaobin/deep-high-resolution-net.pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月23日(星期四) 上午10:41
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [leoxiaobin/deep-high-resolution-net.pytorch] train my own dataset , but AP  is very low (#264)





 
hi,
 how you used this architecture on a custom dataset?
 
Thanks
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by khandriod** [comment]
Thanks for your reply. I do not have labeled data. So, should I use pre-trained model weights for my dataset? 
and where are the results stored (images with pose skeleton)?

**Comment by ing-NT** [comment]
How can you train your own data without&nbsp; the label??
train result stored:output/coco/pose_hrnet/w32_256x192_adam_lr1e-3/results







------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "leoxiaobin/deep-high-resolution-net.pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月23日(星期四) 下午2:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [leoxiaobin/deep-high-resolution-net.pytorch] train my own dataset , but AP  is very low (#264)





 
Thanks for your reply. I do not have labeled data. So, should I use pre-trained model weights for my dataset?
 and where are the results stored (images with pose skeleton)?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by khandriod** [comment]
I don't want to train from scratch. That's why I want to use a pre-trained model (transfer learning, if I am not wrong). Can I use pre-trained weight on a custom dataset? 

Thanks

**Comment by ing-NT** [comment]
Of course you can





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "leoxiaobin/deep-high-resolution-net.pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月23日(星期四) 下午3:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [leoxiaobin/deep-high-resolution-net.pytorch] train my own dataset , but AP  is very low (#264)





 
I don't want to train from scratch. That's why I want to use a pre-trained model (transfer learning, if I am not wrong). Can I use pre-trained weight on a custom dataset?
 
Thanks
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by khandriod** [comment]
could you please write down the steps that I should follow in order to achieve the goal?
my goal is to use HRNET to estimate the human pose on my dataset using pre-trained weights.

Thanks

**Comment by Aliiiu** [comment]
@ing-NT Thanks for the feedbacks, please i will like to know if i can train this model, using a pretrained model, on a images with lesser joint label. If not how can i go about it without it causing conflict with the pretrained model.

**Comment by inever789** [comment]
> Make the same data set format as COCO2017 and put it in the same place as the code

Hello, may I ask, if the custom dataset is used, does each picture need to be marked manually and then converted into coco format? Thank you very much.

**Comment by inever789** [comment]
> could you please write down the steps that I should follow in order to achieve the goal? my goal is to use HRNET to estimate the human pose on my dataset using pre-trained weights.
> 
> Thanks

Hello, have you implemented training on custom datasets now? Could you tell me the operation procedure? I'm having the same problem right now. Thank you very much.
